cluster-operator/src/main/java/io/strimzi/operator/cluster/model/nodepools/NodeIdRange.java

```diff
@@ -0,0 +1,200 @@
+        private Integer end;
+
+        /**
+         * Constructs new IdRange
+         *
+         * @param rangeItem     A single node ID or range
+         *
+         * @throws InvalidNodeIdRangeException  Throws InvalidNodeIdRangeException when the range is not valid
+         */
+        public IdRange(String rangeItem) throws InvalidNodeIdRangeException {
+            if (SINGLE_NODE_ID_PATTERN.matcher(rangeItem).matches())    {
+                try {
+                    start = Integer.parseInt(rangeItem);
+                    end = null;
+                } catch (NumberFormatException e)   {
+                    throw new InvalidNodeIdRangeException("Invalid Node ID (not an integer): " + rangeItem, e);
+                }
+            } else if (RANGE_OF_NODE_IDS_PATTERN.matcher(rangeItem).matches())  {
+                String[] startAndEnd = rangeItem.split("\\s*-\\s*");
+
+                if (startAndEnd.length != 2)    {
+                    throw new InvalidNodeIdRangeException("Invalid Node ID range (exactly 2 items must be found): " + rangeItem);
+                }
+
+                try {
+                    start = Integer.parseInt(startAndEnd[0]);
+                    end = Integer.parseInt(startAndEnd[1]);
+                } catch (NumberFormatException e)   {
+                    throw new InvalidNodeIdRangeException("Invalid Node ID (not an integer): " + rangeItem, e);
+                }
+            } else {
+                throw new InvalidNodeIdRangeException("Invalid Node ID range (does not match a single node ID or node ID range): " + rangeItem);
+            }
+        }
+
+        /**
+         * Returns the next node ID
+         *
+         * @return  The next ID or null if this range was already depleted
+         */
+        public Integer getNextNodeId()   {
+            if (start == null)  {
+                // Starts is null => this range is empty -> return null to indicate that this range was exhausted and should be removed
+                return null;
+            } else if (end == null)    {
+                // Start is not null but end is null => this is a single node id -> return start and set it to null since we are empty now
+                Integer returnValue = start;
+                start = null;
+                return returnValue;
+            } else {
+                // we have a non-empty range
+                if (start.equals(end))   {
+                    // Start is the end => return start and set it to null since we will be empty soon
+                    Integer returnValue = start;
+                    start = null;
+                    end = null;
+                    return returnValue;
+                } else if (start > end) {
+                    // Start is bigger than end => return start and decrement it (the range is going down)
+                    Integer returnValue = start;
+                    start--;
+                    return returnValue;
```

Comment:
related to the previous question, but how much would it be clear to user that in case of, for example, [100-20] it's using 100 and then becoming [99-20]. I mean it's a smart move because anyway you are using the right range, from 20 to 100, but not sure how much it would be clear, instead of just pushing the user to declare a correct [20-100].

Reply:
Please see the answer above.